### ISSUE TITLE
Certbot-Nginx parser failed if found rewrite rule

### ISSUE THREAD

**Issue by DeKaN** [certbot/certbot#3991]
certbot-nginx 0.9.3
OS: Ubuntu 14.04
Command: `~/certbot-auto --nginx -d domain.ru -d www.domain.ru`
Actual output: `Cannot find a VirtualHost matching domain domain.ru.`

Domain config:
```
server {
    listen 80;
    root /var/www/domain.ru;
    access_log /var/log/nginx/domain.ru.access.log;
    error_log /var/log/nginx/domain.ru.error.log warn;
    server_name domain.ru www.domain.ru;
    charset utf-8;
    gzip on;
    gzip_static on;
    gzip_proxied any;
    gzip_types application/json application/x-javascript text/css;
    gzip_min_length 1024;
    client_max_body_size 5M;

    location = /robots.txt {
        return 200 "User-Agent: *\nDisallow: /\n";
    }

    # parser problem here
    rewrite ^/p/([-+]?\d+\.\d+,[-+]?\d+\.\d+),(\d+)$ https://domain.com/?pll=$1&z=$2&ll=$1 permanent;

    location / {
        uwsgi_pass unix:///tmp/domain.sock;
        include uwsgi_params;
        # the uwsgi_params file you installed
    }
}
```

**Comment by ohemorange** [comment]
This is an error in nginxparser.py. @schoen, you like regexes!

For now, @DeKaN, you could comment out that line, run certbot, then uncomment the line, or use a different plugin to get the cert then install it manually.

**Comment by DeKaN** [comment]
This regular expression for `rewrite` in nginx
Rewrite may be in server and location blocks

```
from pyparsing import oneOf
rewrite_statement = space + Literal("rewrite") + space + CharsNotIn("{};" + string.whitespace) + space + CharsNotIn("{};" + string.whitespace) + Optional(oneOf("last break redirect permanent")) + semicolon
```

**Comment by ohemorange** [comment]
This should be fixed by #4296.

**Comment by ohemorange** [comment]
#4296 is in, so closing this.
